transitions/generate_patient_id_on_patients.js

```diff
@@ -0,0 +1,11 @@
+var transitionUtils = require('./utils');
+
+module.exports = {
+  filter: function(doc) {
+    return doc.type === 'person' &&
```

Comment:
After further discussion on hangouts it's determined that it's not necessary/feasible to make changes to how we process the short code or `patient_id` property, so much of the discussion above can be ignored.  But there are still two issues/things to be done to make this PR stable.  The main problem here is that now every person will have a short code or `patient_id` property and we will run out of IDs pretty quickly (maximum at 10000 [still need to confirm](https://github.com/medic/medic-sentinel/blob/master/lib/ids.js)) with a default id format setting of `1111`.  So in order to release this into the wild we'll need to be able to increase that length automatically, that setting will mostly go away (or be moved to the actual transition config only for advanced users), and be managed automatically within sentinel.  Whenever sentinel hits a retry limit (maybe 5 or something) then it will increment the length of that ID automatically by one.  Otherwise sentinel would quickly get stuck on this transition on instances with more than 10k people and all transitions would stop working.